File: fibonacci.c
#include <stdio.h>

int fibonacci(int n) {
  if (n < 0) {
    return -1; // Error case for negative input
  } else if (n == 0) {
    return 0;
  } else if (n == 1) {
    return 1;
  } else {
    int a = 0;
    int b = 1;
    int temp;
    for (int i = 2; i <= n; i++) {
      temp = a + b;
      a = b;
      b = temp;
    }
    return b;
  }
}

int main() {
  // Example usage
  printf("fibonacci(0) = %d\n", fibonacci(0));
  printf("fibonacci(1) = %d\n", fibonacci(1));
  printf("fibonacci(2) = %d\n", fibonacci(2));
  printf("fibonacci(10) = %d\n", fibonacci(10));
  printf("fibonacci(-1) = %d\n", fibonacci(-1)); // Test error case
  return 0;
}

File: Fibonacci.cs
using System;

public class FibonacciSequence
{
    public static int Fibonacci(int n)
    {
        if (n < 0)
        {
            throw new ArgumentOutOfRangeException(nameof(n), "Input must be a non-negative integer.");
        }
        if (n == 0)
        {
            return 0;
        }
        if (n == 1)
        {
            return 1;
        }

        int a = 0;
        int b = 1;
        for (int i = 2; i <= n; i++)
        {
            int temp = a + b;
            a = b;
            b = temp;
        }
        return b;
    }

    public static void Main(string[] args)
    {
        // Example usage
        Console.WriteLine($"Fibonacci(0) = {Fibonacci(0)}");
        Console.WriteLine($"Fibonacci(1) = {Fibonacci(1)}");
        Console.WriteLine($"Fibonacci(2) = {Fibonacci(2)}");
        Console.WriteLine($"Fibonacci(10) = {Fibonacci(10)}");

        try
        {
            Fibonacci(-1);
        }
        catch (ArgumentOutOfRangeException e)
        {
            Console.WriteLine($"Error: {e.Message}");
        }
    }
}
